Sources/TuistGenerator/Generator/LinkGenerator.swift

```diff
@@ -93,24 +93,37 @@ final class LinkGenerator: LinkGenerating {
         let embeddableFrameworks = try graph.embeddableFrameworks(path: path, name: target.name)
         let linkableModules = graph.linkableDependencies(path: path, name: target.name)
 
-        try setupSearchAndIncludePaths(target: target,
-                                       pbxTarget: pbxTarget,
-                                       sourceRootPath: sourceRootPath,
-                                       path: path,
-                                       graph: graph,
-                                       linkableModules: linkableModules)
-
-        try generateEmbedPhase(dependencies: embeddableFrameworks,
-                               target: target,
-                               pbxTarget: pbxTarget,
-                               pbxproj: pbxproj,
-                               fileElements: fileElements,
-                               sourceRootPath: sourceRootPath)
-
-        try generateLinkingPhase(dependencies: linkableModules,
-                                 pbxTarget: pbxTarget,
-                                 pbxproj: pbxproj,
-                                 fileElements: fileElements)
+        if !linkableModules.isEmpty {
+            try setupSearchAndIncludePaths(target: target,
+                                           pbxTarget: pbxTarget,
+                                           sourceRootPath: sourceRootPath,
+                                           path: path,
+                                           graph: graph,
+                                           linkableModules: linkableModules)
+        }
```

Comment:
looking through, I think it's safe to continue setting up the search and include paths as those are no ops incase there aren't any to setup and they don't add any extra build phase either way.